src/ServiceBus.AttachmentPlugin/AzureStorageAttachment.cs

```diff
@@ -46,14 +46,16 @@ public override async Task<Message> BeforeMessageSend(Message message)
             try
             {
                 // Will only work for Shared Key or Account SAS. For Container SAS will throw an exception.
-                if (! await container.ExistsAsync().ConfigureAwait(false))
+                if (! await container.ExistsAsync().ConfigureAwait(false)) 
                 {
                     await container.CreateIfNotExistsAsync().ConfigureAwait(false);
                 }
             }
-            catch (StorageException)
+            catch (StorageException exception)
             {
                 // swallow in case a container SAS is used
+                if (exception.RequestInformation.HttpStatusCode != (int)System.Net.HttpStatusCode.Forbidden)
+                    throw exception;
```

Comment:
@rmt-2021, if I understand the change correctly, the caught exception should be rethrown here, but it should be rethrown, not throw a new exception by re-using the captured exception object. If I got it right, I suggest changing the rethrow code to follow the .NET standard exception rethrowing practice.

```suggestion
                if (exception.RequestInformation.HttpStatusCode != (int)System.Net.HttpStatusCode.Forbidden)
                {
                    throw;
                }
```

Reply:
If you agree, you can `Commit suggestion,` and I'll merge the PR.